inmem_transport.go

```diff
@@ -139,20 +139,27 @@ func (i *InmemTransport) makeRPC(target ServerAddress, args interface{}, r io.Re
 	}
 
 	// Send the RPC over
-	respCh := make(chan RPCResponse)
-	peer.consumerCh <- RPC{
+	respCh := make(chan RPCResponse, 1)
+	rpc := RPC{
 		Command:  args,
 		Reader:   r,
 		RespChan: respCh,
 	}
+	timer := time.After(timeout)
```

Comment:
Since
- I didn't make this any worse in this PR (the `time.After` was there on line 155 before) and
- there are 13 other occurrences of `time.After` in the non-test code in the library,

I'm inclined not to address this as part of this PR.

@rogpeppe, if dangling timers are an issue, could you file this in a separate issue to audit the timer use in the entire library? And more importantly, get the documentation on https://golang.org/pkg/time/#After updated to describe the issue and its severity.